src/demo-app/table/table-demo.ts

```diff
@@ -16,23 +18,41 @@ const properties = ['id', 'name', 'progress', 'color'];
   styleUrls: ['table-demo.css'],
 })
 export class TableDemo {
-  dataSource: PersonDataSource | null;
-  displayedColumns: UserProperties[] = [];
+  dataSource: MdTableDataSource<UserData>|null;
+  displayedColumns: UserProperties[] = ['userId', 'userName', 'progress', 'color'];
   trackByStrategy: TrackByStrategy = 'reference';
   changeReferences = false;
   highlights = new Set<string>();
 
+  filter = new FormControl();
+
   dynamicColumnDefs: any[] = [];
   dynamicColumnIds: string[] = [];
 
   @ViewChild(MdPaginator) _paginator: MdPaginator;
 
   @ViewChild(MdSort) sort: MdSort;
 
-  constructor(public _peopleDatabase: PeopleDatabase) { }
+  constructor(public _peopleDatabase: PeopleDatabase) {
+    this.dataSource = new MdTableDataSource(this._peopleDatabase.data);
+    this.dataSource.dataAccessor = (data: UserData, property: string) => {
+      switch (property) {
+        case 'userId': return +data.id;
+        case 'userName': return data.name;
+        case 'progress': return +data.progress;
+        case 'color': return data.color;
+        default: return '';
+      }
+    };
+    this.dataSource.filterTermAccessor = (data: UserData) => data.name;
+    this.filter.valueChanges.subscribe(filter => this.dataSource!.filter = filter);
+  }
 
-  ngOnInit() {
-    this.connect();
+  ngAfterViewInit() {
+    // Needs to be set up after the view is initialized since the data source will look at the sort
+    // and paginator's initial values to know what data should be rendered.
+    this.dataSource!.sort = this.sort;
+    this.dataSource!.paginator = this._paginator;
```

Comment:
This looks awesome and _very_ user friendly! `MdTableDataSource`'s properties seem to document themselves, but I think using ViewChild and setting the instances in AfterViewInit will trip up some newcomers. IMO this little bit necessitates an example. I'd suggest using an "out of the way" data array like the Basic Example to emphasize how easy this is.

Reply:
Totally, I'm racking my brain to think of ways we can help devs avoid this but I can't think of anything. I wish I had access to the view ref in the data source and know whether or not they are passing it in at the right time. Maybe I should throw an error if they pass in an empty sort or paginator, saying that perhaps they are assigning it at the wrong time

Reply:
The error would be nice, but I'm afraid it's a valid - though rare - use case to initially pass in a paginator and then try to disable it later during the table's life. If you could get around that, the more descriptive errors are, the better!

Reply:
Here's an idea of a more magical version. MdSort can inject itself into the data source by injecting the table instance (assumes it is on the table element). MdPaginator can create an input that accepts the data source. This way, the user doesn't need to know about view init.

Check it out here: https://github.com/angular/material2/commit/097a2ffc6c5d638400d9ba38ad2fb5eedd14aeec

@jelbourn I know you are avoiding magic, think this goes too far? Or are we not going far enough... =D

Reply:
Hah that's pretty slick! 🔮 I still think it'd require an example since passing the data source to the paginator is not an immediately obvious thing to do.. I prefer the original; the AfterViewInit part is plenty straight forward once you've done it once. I just don't think everyone will realize that's the correct approach without an example.

Reply:
I'll defer to your judgment that people would get it after seeing an example. I imagine we'll have some canonical StackOverflow question that addresses the issue people would see. Would prefer not to have such tightly coupled magic anyways. If we end up seeing that its an issue, we can always add it into MdSort and MdPaginator later too